processing/src/main/java/org/apache/druid/segment/join/Joinables.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.join;
+
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.segment.column.ColumnHolder;
+
+import javax.annotation.Nullable;
+
+/**
+ * Utility methods for working with {@link Joinable} related classes.
+ */
+public class Joinables
+{
+  /**
+   * Checks that "prefix" is a valid prefix for a join clause (see {@link JoinableClause#getPrefix()}) and, if so,
+   * returns it. Otherwise, throws an exception.
+   */
+  public static String validatePrefix(@Nullable final String prefix)
+  {
+    if (prefix == null) {
+      throw new IAE("Join clause cannot have null prefix");
+    } else if (isPrefixedBy(ColumnHolder.TIME_COLUMN_NAME, prefix) || ColumnHolder.TIME_COLUMN_NAME.equals(prefix)) {
+      throw new IAE(
+          "Join clause cannot have prefix[%s], since it would shadow %s",
+          prefix,
+          ColumnHolder.TIME_COLUMN_NAME
+      );
+    } else {
+      return prefix;
+    }
+  }
+
+  public static boolean isPrefixedBy(final String columnName, final String prefix)
```

Comment:
`@NonNull` annotations for the parameters?

Reply:
I think I'd prefer `@EverythingIsNonnullByDefault` (which is how I prefer to write/read code usually anyway). We use it in a few other packages in Druid. Maybe we can add that here in a future patch?

Reply:
Works for me! I haven't gotten in the habit of looking out for those annotations yet, but I love the idea of NonnullBy default!